### PR TITLE
DEPL-9885 fixed Failing tests, added corner cases involving spaces in…

### PR DESCRIPTION
**Pull request by sameerarora**
… arguments to batch script in windows. Added tests.

### CODE REVIEW

File: src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java [hunk 1]
```diff
@@ -24,9 +24,18 @@
 
 public class WindowsCommandLineArgsSanitizer {
 
-    private static final char[] SPECIAL_CHARS = "|<>&^\r\n".toCharArray();
+    private static final char[] SPECIAL_CHARS = " |<>&^\r\n".toCharArray();
```
Comment (hierynomus) [inline src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java]: Why is the space added to the SPECIAL_CHARS? The space should not be escaped with a caret, right?

Comment (sameerarora) [inline src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java]: yes , I have removed it, since space is getting checked before everything else , this would have never run escape logic for space with a caret.

Comment (sameerarora) [inline src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java]: sorry I moved in too fast , this only is needed to determine if the Argument to be escaped has any special characters or not , space is never escaped by a caret in the sanitize method

Comment (hierynomus) [inline src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java]: Ok, now the 1 million dollar question. If a string contains spaces _and_ special characters, we're now not escaping any carets. Isn't it good enough then to _always_ put quotes around the string?

Comment (sameerarora) [inline src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java]: No, it breaks in certain scenarios like a winrs -r command run from a localhost as winrs proxy. This one is a specific case.

Comment (hierynomus) [inline src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java]: But how is that solved now? Because you don't escape anything if there's a space. Isn't a space escapable using a caret?

Op 27 jun. 2016 11:26 a.m. schreef Sameer Arora <notifications@github.com>:

In src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java<https://github.com/xebialabs/overthere/pull/180#discussion_r68546485>:

> @@ -24,9 +24,18 @@
>
>  public class WindowsCommandLineArgsSanitizer {
>
> -    private static final char[] SPECIAL_CHARS = "|<>&^\r\n".toCharArray();
> +    private static final char[] SPECIAL_CHARS = " |<>&^\r\n".toCharArray();


No, it breaks in certain scenarios like a winrs -r command run from a localhost as winrs proxy. This one is a specific case.

Comment (sameerarora) [inline src/main/java/com/xebialabs/overthere/util/WindowsCommandLineArgsSanitizer.java]: That is solved in the localconnection class

**Comment by seveneves** [comment]
@sameerarora @hierynomus what's the status of this PR. Can it be merged?

**Comment by sameerarora** [comment]
@vpartington @hierynomus any more comments on this ?

**Comment by sameerarora** [comment]
we run the commands on a LocalConnection by prefixing it with  cmd /c <Your_Command_Here>

**Comment by hierynomus** [comment]
ok, let's merge this for now...
